.github/workflows/robot/internal/bot/queue_test.go

```diff
@@ -0,0 +1,127 @@
+		canUpdate bool
+	}{
+		{
+			desc: "fork-skip",
+			pr: github.PullRequest{
+				Author:        "foo",
+				Repository:    "bar",
+				UnsafeHeadRef: "baz/qux",
+				HeadSHA:       "0000000000000000000000000000000000000000000000000000000000000001",
+				UnsafeBaseRef: "master",
+				BaseSHA:       "0000000000000000000000000000000000000000000000000000000000000001",
+				Fork:          true,
+				AutoMerge:     true,
+			},
+			cache: map[string]string{
+				"master": "0000000000000000000000000000000000000000000000000000000000000001",
+			},
+			canUpdate: false,
+		},
+		{
+			desc: "non-master-skip",
+			pr: github.PullRequest{
+				Author:        "foo",
+				Repository:    "bar",
+				UnsafeHeadRef: "baz/qux",
+				HeadSHA:       "0000000000000000000000000000000000000000000000000000000000000001",
+				UnsafeBaseRef: "branch/v0",
+				BaseSHA:       "0000000000000000000000000000000000000000000000000000000000000001",
+				Fork:          false,
+				AutoMerge:     true,
+			},
+			cache: map[string]string{
+				"master": "0000000000000000000000000000000000000000000000000000000000000001",
+			},
+			canUpdate: false,
+		},
+		{
+			desc: "no-auto-merge-skip",
+			pr: github.PullRequest{
+				Author:        "foo",
+				Repository:    "bar",
+				UnsafeHeadRef: "baz/qux",
+				HeadSHA:       "0000000000000000000000000000000000000000000000000000000000000001",
+				UnsafeBaseRef: "master",
+				BaseSHA:       "0000000000000000000000000000000000000000000000000000000000000001",
+				Fork:          false,
+				AutoMerge:     false,
+			},
+			cache: map[string]string{
+				"master": "0000000000000000000000000000000000000000000000000000000000000001",
+			},
+			canUpdate: false,
+		},
+		{
+			desc: "up-to-date-skip",
+			pr: github.PullRequest{
+				Author:        "foo",
+				Repository:    "bar",
+				UnsafeHeadRef: "baz/qux",
+				HeadSHA:       "0000000000000000000000000000000000000000000000000000000000000001",
+				UnsafeBaseRef: "master",
+				BaseSHA:       "0000000000000000000000000000000000000000000000000000000000000001",
+				Fork:          false,
+				AutoMerge:     true,
+			},
+			cache: map[string]string{
+				"master": "0000000000000000000000000000000000000000000000000000000000000001",
+			},
+			canUpdate: false,
+		},
+		{
+			desc: "not-up-to-date-update",
+			pr: github.PullRequest{
+				Author:        "foo",
+				Repository:    "bar",
+				UnsafeHeadRef: "baz/qux",
+				HeadSHA:       "0000000000000000000000000000000000000000000000000000000000000001",
+				UnsafeBaseRef: "master",
+				BaseSHA:       "0000000000000000000000000000000000000000000000000000000000000002",
+				Fork:          false,
+				AutoMerge:     true,
+			},
+			cache: map[string]string{
+				"master": "0000000000000000000000000000000000000000000000000000000000000001",
+			},
+			canUpdate: true,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.desc, func(t *testing.T) {
+			ok := canUpdate(test.pr, test.cache)
+			require.Equal(t, ok, test.canUpdate)
```

Comment:
The signature for these assertions is `t, expected, actual`.
Not a huge deal as the test will still fail, but the error message may be confusing if you invert the arguments.

```suggestion
			require.Equal(t, test.canUpdate, ok)
```